### PR TITLE
Trafficgate Bid Adapter: move to OpenRTB Converter

### PR DESCRIPTION
**Pull request by vlaktion**
https://github.com/prebid/prebid.github.io/pull/4606

## Type of change

- [X] Feature
- [X] Does this change affect user-facing APIs or examples documented on http://prebid.org?
   Yes

## Description of change
Rewrite adapter on OpenRTB Converter for receiving an OpenRTB request on the server and response with BidResponse OpenRTB

- test parameters for validating bids:
Banner 300x250px
```
{
  bidder: 'trafficgate',
  params: {
    placementId: '16',
    host: 'bidscube',
    test: 1
  }
}
```

Video 640x480px
```
{
   bidder: 'trafficgate',
    params: {
      placementId: '10',
      host: 'bidscube',
      test: 1
    }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page. -->
var adUnits = [{
    code: 'div-gpt-ad-1460505748561-0',
    sizes: [[300, 250]],  // a display size
    mediaTypes: {
    banner: {
      sizes: [[300, 250]],
    }
    },
    bids: [{
    bidder: 'trafficgate',
    params: {
      placementId: '16',
      host: 'bidscube',
      test: 1
    }
    }]
}];

### CODE REVIEW

**Comment by robertrmartinez** [comment]
@vlaktion 

Hi I will review this soon! Sorry for the delay.

**Comment by vlaktion** [comment]
Hello. When this PR will be merged and released?

File: modules/trafficgateBidAdapter.js [hunk 1]
```diff
@@ -1,95 +1,155 @@
...
+  request(buildRequest, imps, bidderRequest, context) {
+    const req = buildRequest(imps, bidderRequest, context);
+    mergeDeep(req, {
+      at: 1,
+    })
+    const bid = context.bidRequests[0];
+    if (bid.params.coppa) {
+      deepSetValue(req, 'regs.coppa', 1);
+    }
+    if (bid.params.doNotTrack) {
```
Comment (robertrmartinez) [inline modules/trafficgateBidAdapter.js:50]: also this is something prebid willl set based on several checks here:
https://github.com/prebid/Prebid.js/blob/master/src/fpd/enrichment.js#L99
https://github.com/prebid/Prebid.js/blob/master/src/utils.js#L938

So I think not overwriting is the right thing to do here

**Review by robertrmartinez — CHANGES_REQUESTED** [review]
sorry took so long, looks fine just a couple questions

File: modules/trafficgateBidAdapter.js [hunk 2]
```diff
@@ -1,95 +1,155 @@
...
-      const item = body[i]
-      if (isBidResponseValid(item)) {
-        response.push(item)
+  request(buildRequest, imps, bidderRequest, context) {
+    const req = buildRequest(imps, bidderRequest, context);
+    mergeDeep(req, {
+      at: 1,
+    })
+    const bid = context.bidRequests[0];
+    if (bid.params.coppa) {
```
[the remark this answers is not in the document]
Comment (vlaktion) [reply modules/trafficgateBidAdapter.js]: Okey. I removed it.

File: modules/trafficgateBidAdapter.js (hunk 1 above)
Comment (vlaktion) [reply modules/trafficgateBidAdapter.js]: I removed this too.

**Comment by vlaktion** [comment]
Done!

**Review by robertrmartinez — APPROVED** [review]

**Pull request merged**